### PR TITLE
Use using instead of typedef [gpu]

### PR DESCRIPTION
**Pull request by SunBlack**
Changes are done by: `run-clang-tidy -header-filter='.' -checks='-,modernize-use-using' -fix`

### CODE REVIEW

File: gpu/kinfu/tools/kinfu_app.cpp [hunk 1]
```diff
@@ -959,20 +958,20 @@ struct KinFuApp
...
   {   
     using namespace openni_wrapper;
-    typedef boost::shared_ptr<DepthImage> DepthImagePtr;
-    typedef boost::shared_ptr<Image> ImagePtr;
+    using DepthImagePtr = boost::shared_ptr<DepthImage>;
+    using ImagePtr = boost::shared_ptr<Image>;
         
-    std::function<void (const ImagePtr&, const DepthImagePtr&, float constant)> func1_dev = boost::bind (&KinFuApp::source_cb2_device, this, _1, _2, _3);
-    std::function<void (const DepthImagePtr&)> func2_dev = boost::bind (&KinFuApp::source_cb1_device, this, _1);
+    boost::function<void (const ImagePtr&, const DepthImagePtr&, float constant)> func1_dev = boost::bind (&KinFuApp::source_cb2_device, this, _1, _2, _3);
```
Comment (taketwo) [inline gpu/kinfu/tools/kinfu_app.cpp]: Are we switching back to `boost`? :D

File: gpu/kinfu_large_scale/tools/kinfuLS_app.cpp [hunk 2]
```diff
@@ -979,12 +978,12 @@ struct KinFuLSApp
...
     using namespace openni_wrapper;
-    typedef boost::shared_ptr<DepthImage> DepthImagePtr;
-    typedef boost::shared_ptr<Image>      ImagePtr;
+    using DepthImagePtr = boost::shared_ptr<DepthImage>;
+    using ImagePtr = boost::shared_ptr<Image>;
 
-    std::function<void (const ImagePtr&, const DepthImagePtr&, float constant)> func1 = boost::bind (&KinFuLSApp::source_cb2, this, _1, _2, _3);
-    std::function<void (const DepthImagePtr&)> func2 = boost::bind (&KinFuLSApp::source_cb1, this, _1);
-    std::function<void (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr&) > func3 = boost::bind (&KinFuLSApp::source_cb3, this, _1);
+    boost::function<void (const ImagePtr&, const DepthImagePtr&, float constant)> func1 = boost::bind (&KinFuLSApp::source_cb2, this, _1, _2, _3);
```
Comment (taketwo) [inline gpu/kinfu_large_scale/tools/kinfuLS_app.cpp]: Same here

File: gpu/kinfu_large_scale/tools/record_maps_rgb.cpp [hunk 3]
```diff
@@ -258,9 +257,9 @@ void
 grabAndSend ()
 {
   pcl::Grabber* interface = new pcl::OpenNIGrabber ();
-  //std::function<void (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr& )> f = boost::bind(&grabberCallBack, _1);
+  //boost::function<void (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr& )> f = boost::bind(&grabberCallBack, _1);
```
Comment (taketwo) [inline gpu/kinfu_large_scale/tools/record_maps_rgb.cpp]: And here

File: gpu/people/tools/people_tracking.cpp [hunk 4]
```diff
@@ -14,17 +15,15 @@
...
 #include <pcl/people/trees/tree_live.h>
+//#include <pcl/filters/passthrough.h>
+//#include <pcl/common/time.h>
 #include <pcl/io/openni_grabber.h>
 #include <pcl/visualization/cloud_viewer.h>
 #include <pcl/console/parse.h>
 
-#include <opencv2/opencv.hpp>
-
-#include <functional>
```
Comment (taketwo) [inline gpu/people/tools/people_tracking.cpp]: Don't remove this

File: gpu/people/tools/people_tracking.cpp [hunk 5]
```diff
@@ -59,7 +58,7 @@ class PeopleTrackingApp
     {
       pcl::Grabber* interface = new pcl::OpenNIGrabber();
 
-      std::function<void (const pcl::PointCloud<pcl::PointXYZRGB>::ConstPtr&)> f =
+      boost::function<void (const pcl::PointCloud<pcl::PointXYZRGB>::ConstPtr&)> f =
```
Comment (taketwo) [inline gpu/people/tools/people_tracking.cpp]: Switch back

File: gpu/kinfu/tools/kinfu_app.cpp (hunk 1 above)
Comment (SunBlack) [reply gpu/kinfu/tools/kinfu_app.cpp]: Sorry, it seems I have been a little unfocused during merging. Will fix this later.

Comment (SunBlack) [reply gpu/kinfu/tools/kinfu_app.cpp]: Fixed

**Pull request merged**